src/charm.py

```diff
@@ -26,20 +28,89 @@ def __init__(self, *args: typing.Any) -> None:
         super().__init__(*args)
         self.framework.observe(self.on.config_changed, self.config_service)
 
+    def flask_container(self, require_connected: bool = True) -> Container:
+        """Get the flask application workload container controller.
+
+        Args:
+            require_connected: if set to ``True``, a runtime exception will be raised if the
+                pebble inside the container is not ready.
+
+        Return:
+            The controller of the flask application workload container.
+
+        Raises:
+            RuntimeError: if the pebble inside  the container is not ready while the
+                ``require_connected`` is set to True.
+        """
+        container = self.unit.get_container("flask-app")
+        if require_connected and not container.can_connect():
+            raise RuntimeError("pebble inside flask-app container is not ready")
+        return container
+
     def config_service(self, event: ConfigChangedEvent) -> None:
         """Configure the flask pebble service layer.
 
         Args:
             event: the config-changed event that trigger this callback function.
         """
-        container = self.unit.get_container("flask-app")
+        container = self.flask_container(require_connected=False)
         if not container.can_connect():
             event.defer()
             return
+        service_name = "flask-app"
+        gunicorn_config_path = "/srv/flask/gunicorn.conf.py"
```

Comment:
this looks like a good candidate for a separate module? Any thoughts? @gregory-schiano as well

Reply:
Yes definitely, we need a state and a service to manage the file generation from the state and managing the generated file

Reply:
I refactored the code to conformed with the standard, thank you.